pyspark_ai/tool.py

```diff
@@ -101,13 +103,70 @@ async def _arun(self, *args: Any, **kwargs: Any) -> Any:
         raise NotImplementedError("ListTablesSqlDbTool does not support async")
 
 
+class LRUVectorStore:
+    """Implements an LRU policy to enforce a max storage space for vector file storage."""
+
+    def __init__(self, vector_file_dir: str, max_size: float = 16) -> None:
+        # by default, max_size = 16 GB
+        self.files: OrderedDict[str, float] = OrderedDict()
+        self.vector_file_dir = vector_file_dir
+        # represent in bytes to prevent floating point errors
+        self.max_bytes = max_size * 1e9
+        self.current_size = 0
+
+        # initialize the file cache, if vector_file_dir exists
+        # existing files will get evicted in reverse-alphabetical order
+        # TODO: write LRU to disk, to evict existing files in LRU order
+        if os.path.exists(self.vector_file_dir):
+            for file in os.listdir(self.vector_file_dir):
+                file_full_path = os.path.join(self.vector_file_dir, file)
+                file_size = LRUVectorStore.get_file_size_bytes(file_full_path)
+                self.current_size += file_size
+                self.files[file_full_path] = file_size
+
+    @staticmethod
+    def get_file_size_bytes(file_path: str) -> float:
+        return os.path.getsize(file_path)
+
+    @staticmethod
+    def get_storage(vector_file_dir: str) -> float:
+        # calculate current storage space of vector files, in bytes
+        size = 0
+        for path, dirs, files in os.walk(vector_file_dir):
+            for f in files:
+                fp = os.path.join(path, f)
+                size += LRUVectorStore.get_file_size_bytes(fp)
+        return size
+
+    def access(self, file_path: str) -> None:
+        # move accessed key to end of LRU cache
+        if file_path in self.files:
+            self.files.move_to_end(file_path)
+
+    def add(self, file_path: str) -> None:
+        # remove file path if max storage size exceeded, else add
+        curr_file_size = LRUVectorStore.get_file_size_bytes(file_path)
```

Comment:
What if curr_file_size is larger than the max size limit? It will be added to the store and then everything will be deleted..